File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppJava/app/src/main/java/pl/udu/uwr/pum/verynobleappjava/adapters/laureatenobelprizes/NobelPrizeViewHolder.java
package pl.udu.uwr.pum.verynobleappjava.adapters.laureatenobelprizes;

import androidx.recyclerview.widget.RecyclerView;

import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.nobelprize.NobelPrize;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.nobelprize.affiliation.Affiliation;
import pl.udu.uwr.pum.verynobleappjava.databinding.LaureateNobelPrizeRvItemBinding;

public class NobelPrizeViewHolder extends RecyclerView.ViewHolder {
    private final LaureateNobelPrizeRvItemBinding binding;
    public NobelPrizeViewHolder(LaureateNobelPrizeRvItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(NobelPrize item){
        binding.categoryFullNameTextView.setText(item.categoryFullName.en);
        binding.portionTextView.setText(item.portion);
        binding.yearTextView.setText(item.awardYear);
        binding.motivationTextView.setText(item.motivation.en);
        binding.affiliationsTextView.setText(affiliations(item));
    }

    private String affiliations(NobelPrize item){
        StringBuilder affiliations = new StringBuilder();
        if (item.affiliations != null) {
            for (Affiliation affiliation : item.affiliations) {
                if (affiliation.name != null && affiliation.name.en != null)
                    affiliations.append(affiliation.name.en).append("\n");
            }
        } else
            affiliations.append("not specified");
        return affiliations.toString();
    }
}

File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppKotlin/app/src/main/java/pl/udu/uwr/pum/verynobleappkotlin/adapters/laureatenobleprizes/NobelPrizeViewHolder.kt
package pl.udu.uwr.pum.verynobleappkotlin.adapters.laureatenobleprizes

import androidx.recyclerview.widget.RecyclerView
import pl.udu.uwr.pum.verynobleappkotlin.data.laureateresponse.nobelprize.NobelPrize
import pl.udu.uwr.pum.verynobleappkotlin.databinding.LaureateNobelPrizeRvItemBinding

class NobelPrizeViewHolder(private val binding: LaureateNobelPrizeRvItemBinding)
    : RecyclerView.ViewHolder(binding.root) {
    fun bind(item: NobelPrize){
        binding.yearTextView.text = item.awardYear
        binding.portionTextView.text = item.portion
        binding.motivationTextView.text = item.motivation.en
        binding.categoryFullNameTextView.text = item.categoryFullName.en
        binding.affiliationsTextView.text = affiliations(item)
    }

    private fun affiliations(item: NobelPrize): String{
        val affiliations = StringBuilder()
        item.affiliations?.forEach { affiliations.append(it.name.en).append("\n") }
        return affiliations.toString()
    }
}